recirq/kpz/experiment.py

```diff
@@ -0,0 +1,538 @@
+            @ (self.transferred_magnetization_vals - self.mean) ** 3
+            / self.variance ** (3 / 2)
+        )
+
+    def _kurtosis(self) -> float:
+        return (
+            self.transferred_magnetization_probs
+            @ (self.transferred_magnetization_vals - self.mean) ** 4
+            / self.variance**2
+            - 3
+        )
+
+    def _mean_excluding_i(self, i: int) -> float:
+        p = np.mean(
+            np.delete(self.transferred_magnetization_probs_all, i, axis=0), axis=0
+        )
+        return p @ self.transferred_magnetization_vals
+
+    def _variance_excluding_i(self, i: int) -> float:
+        p = np.mean(
+            np.delete(self.transferred_magnetization_probs_all, i, axis=0), axis=0
+        )
+        mean_i = p @ self.transferred_magnetization_vals
+        return p @ (self.transferred_magnetization_vals - mean_i) ** 2
+
+    def _skew_excluding_i(self, i: int) -> float:
+        p = np.mean(
+            np.delete(self.transferred_magnetization_probs_all, i, axis=0), axis=0
+        )
+        mean_i = p @ self.transferred_magnetization_vals
+        variance_i = p @ (self.transferred_magnetization_vals - mean_i) ** 2
+        return (
+            p
+            @ (self.transferred_magnetization_vals - mean_i) ** 3
+            / variance_i ** (3 / 2)
+        )
+
+    def _kurtosis_excluding_i(self, i: int) -> float:
+        p = np.mean(
+            np.delete(self.transferred_magnetization_probs_all, i, axis=0), axis=0
+        )
+        mean_i = p @ self.transferred_magnetization_vals
+        variance_i = p @ (self.transferred_magnetization_vals - mean_i) ** 2
+        return (
+            p @ (self.transferred_magnetization_vals - mean_i) ** 4 / variance_i**2
+            - 3
+        )
+
+    def jackknife_mean(self) -> float:
+        """Compute the statistical uncertainty using the remove-one jackknife"""
+        if self.num_initial_states == 1:
+            return 0
+        mean_i = [self._mean_excluding_i(i) for i in range(self.num_initial_states)]
+        return np.std(mean_i) * np.sqrt(self.num_initial_states - 1)
+
+    def jackknife_variance(self) -> float:
+        """Compute the statistical uncertainty using the remove-one jackknife"""
+        if self.num_initial_states == 1:
+            return 0
+        variance_i = [
+            self._variance_excluding_i(i) for i in range(self.num_initial_states)
+        ]
+        return np.std(variance_i) * np.sqrt(self.num_initial_states - 1)
+
+    def jackknife_skew(self) -> float:
+        """Compute the statistical uncertainty using the remove-one jackknife"""
+        if self.num_initial_states == 1:
+            return 0
+        skew_i = [self._skew_excluding_i(i) for i in range(self.num_initial_states)]
+        return np.std(skew_i) * np.sqrt(self.num_initial_states - 1)
+
+    def jackknife_kurtosis(self) -> float:
+        """Compute the statistical uncertainty using the remove-one jackknife"""
```

Comment:
All of these functions have the docstring "Compute the statistical uncertainty using the remove-one jackknife".  Should we fix that?  Same for the below class.

Reply:
Implemented